### PR TITLE
Using `@unknown default` to ensure we handle future cases

### PR DESCRIPTION
**Pull request by NachoSoto**
I like to take some credit in this https://twitter.com/NachoSoto/status/943880484978769921?s=20

### CODE REVIEW

File: Purchases/Error Handling/ErrorUtils.swift [hunk 1]
```diff
@@ -347,4 +320,44 @@ private extension ErrorUtils {
         return errorWithUserInfo
     }
 
+    private static func logErrorIfNeeded(_ code: ErrorCode) {
+        switch code {
+        case .networkError,
+                .unknownError,
```
Comment (taquitos) [inline Purchases/Error Handling/ErrorUtils.swift:326]: formatting is weird, I wonder why they got pushed out like this

File: Purchases/Error Handling/ErrorUtils.swift [hunk 2]
```diff
@@ -347,4 +320,44 @@ private extension ErrorUtils {
...
+                .configurationError,
+                .unsupportedError,
+                .emptySubscriberAttributes,
+                .productDiscountMissingIdentifierError,
+                .missingAppUserIDForAliasCreationError,
+                .productDiscountMissingSubscriptionGroupIdentifierError:
+            break
+
+        @unknown default:
+            break
```
Comment (taquitos) [inline Purchases/Error Handling/ErrorUtils.swift:360]: Maybe we still want to log something, pointing out we don't know.

File: Purchases/Public/IntroEligibility.swift [hunk 3]
```diff
@@ -83,8 +83,10 @@ private extension IntroEligibilityStatus {
             return "Eligible for trial or introductory price."
         case .ineligible:
             return "Not eligible for trial or introductory price."
-        default:
-            return "Status indeterminate."
+
+        case .unknown: fallthrough
```
Comment (taquitos) [inline Purchases/Public/IntroEligibility.swift:87]: TIL `fallthrough`

File: Purchases/Error Handling/ErrorUtils.swift (hunk 1 above)
Comment (NachoSoto) [reply Purchases/Error Handling/ErrorUtils.swift:326]: Xcode auto-formatting was doing it like this, couldn't figure out why.

File: Purchases/Public/IntroEligibility.swift (hunk 3 above)
Comment (NachoSoto) [reply Purchases/Public/IntroEligibility.swift:87]: No fallthrough by default is one of those small things that makes Swift a much better language.

File: Purchases/Error Handling/ErrorUtils.swift (hunk 2 above)
Comment (NachoSoto) [reply Purchases/Error Handling/ErrorUtils.swift:360]: 👍

Comment (NachoSoto) [reply Purchases/Error Handling/ErrorUtils.swift:360]: Actually this was like this already, not all logs were being logged. I didn't realize I should be putting some of these in the right category, then I'll log something in the `default`.

File: Purchases/Error Handling/ErrorUtils.swift (hunk 1 above)
Comment (taquitos) [reply Purchases/Error Handling/ErrorUtils.swift:326]: Ugh, I hate it, but I think we should stick with whatever the default is for Xcode

**Review by taquitos — APPROVED** [review]
🕺🐐

File: Purchases/Error Handling/ErrorUtils.swift (hunk 1 above)
Comment (NachoSoto) [reply Purchases/Error Handling/ErrorUtils.swift:326]: Yeah not worth indenting manually I think..

**Pull request merged**